cirq-core/cirq/sim/act_on_args.py

```diff
@@ -68,13 +73,18 @@ def __init__(
         self._set_qubits(qubits)
         self.prng = prng
         self._log_of_measurement_results = log_of_measurement_results
+        self._ignore_measurement_results = ignore_measurement_results
 
     def _set_qubits(self, qubits: Sequence['cirq.Qid']):
         self._qubits = tuple(qubits)
         self.qubit_map = {q: i for i, q in enumerate(self.qubits)}
 
     def measure(self, qubits: Sequence['cirq.Qid'], key: str, invert_mask: Sequence[bool]):
-        """Adds a measurement result to the log.
+        """Measures the qubits and records to `log_of_measurement_results`.
+
+        Any bitmasks will be applied to the measurement record. If
+        `self._ignore_measurement_results` is set, it dephases instead of
+        measuring.
```

Comment:
We should be explicit that "dephase instead measure" means no measurement result is logged, especially since we now have tools that would allow measurements from a dephasing operation.

Reply:
Done. You almost don't need a measurement gate anymore since it could just be a dephase with a key. (Not that I recommend removing them -- there'll be some more measurement-specific functionality like concat etc in the future). (And I guess we'll need to decide how/whether that works with channel measurements when we do that).

Reply:
> (Not that I recommend removing [measurements] -- there'll be some more measurement-specific functionality like concat etc in the future)

We came to a similar conclusion in the design sync - `MeasurementGate` is here to stay.